gallery/tutorials/tutorials/micrograph_source.py

```diff
@@ -0,0 +1,220 @@
+corrupted_micrograph.micrographs[:].show()
+
+# %%
+# Noise
+# -----
+# By default, no noise corruption is configured. To apply noise, we have to pass them as arguments to the Micrograph Source
+
+# Create our noise using WhiteNoiseAdder
+noise = WhiteNoiseAdder(1e-3, seed=1234)
+
+# Let's add noise to our MicrographSource using the noise_adder argument
+noisy_micrograph = MicrographSource(
+    corrupted_sim,
+    noise_adder=noise,
+    particles_per_micrograph=4,
+    micrograph_size=500,
+    micrograph_count=4,
+    seed=1234,
+)
+
+# Plot the micrographs
+noisy_micrograph.micrographs[:].show()
+
+# %%
+# We can also plot the un-noisy micrographs using the ``clean_micrographs`` accessor
+noisy_micrograph.clean_micrographs[:].show()
+
+# %%
+# Interparticle Distance
+# ----------------------
+# By default, particle distance is set to avoid collisions.
+# We can use this argument to control the minimum distance between particle centers
+# However, setting this argument too large may generate insufficient centers.
+
+# Let's increase the number of particles to show overlap.
+# Create a new simulation to meet the minimum required amount of projections.
+n_particles_per_micrograph = 20
+
+colliding_sim = Simulation(
+    L=64,
+    n=n_particles_per_micrograph * n_micrographs,
+    C=1,
+    amplitudes=1,
+    offsets=0,
+    unique_filters=ctfs,
+)
+
+
+# Set the interparticle distance to 1, which adds at least one pixel of separation between centers.
+# This allows particles to collide but not completely overlap.
+colliding_micrograph = MicrographSource(
+    colliding_sim,
+    interparticle_distance=1,
+    noise_adder=noise,
+    particles_per_micrograph=n_particles_per_micrograph,
+    micrograph_size=500,
+    micrograph_count=n_micrographs,
+)
+
+# Plot the micrographs
+colliding_micrograph.micrographs[:].show()
+
+# %%
+# Boundary
+# --------
+# By default, the boundary is set to half of the particle width , which will completely contain every particle inside the micrograph.
+# Setting ``boundary=0`` will allow particle centers to generate along the edges.
+# Positive values (measured in pixels) move the boundaries inward, while negative values move the boundaries outward.
+
+# Create a micrograph with a negative boundary, allowing particles to generate outward.
+outbound_micrograph = MicrographSource(
+    colliding_sim,
+    boundary=-20,
+    interparticle_distance=1,
+    noise_adder=noise,
+    particles_per_micrograph=n_particles_per_micrograph,
+    micrograph_size=500,
+    micrograph_count=n_micrographs,
+)
+
+# Plot the micrographs
+outbound_micrograph.micrographs[:].show()
+
+# %%
+# Particle IDs
+# ------------
+# Each particle comes from the simulation, and has its own ID relative to its ``Simulation`` and micrograph.
+# Let's choose 4 random numbers, as our global (Simulation) particle IDs from ``test_micrograph=1``.
+test_micrograph = 1
+local_particle_ids = np.random.choice(n_particles_per_micrograph, 4)
```

Comment:
You could set `n_particles = 4` and avoid using `len(local_particle_ids)` in several places below.